apps/aecontract/src/aect_channel_contract.erl

```diff
@@ -0,0 +1,104 @@
+-module(aect_channel_contract).
+-include("aecontract.hrl").
+
+-export([new/6,
+         run_new/5,
+         run/7,
+         get_call/4
+        ]).
+
+new(Owner, Round, VmVersion, Code, Deposit, Trees0) ->
+    %% Create the contract and insert it into the contract state tree
+    %%   The public key for the contract is generated from the owners pubkey
+    %%   and the nonce, so that no one has the private key.
+    Contract        = aect_contracts:new(Owner, Round, VmVersion, Code, Deposit),
+    ContractPubKey  = aect_contracts:pubkey(Contract),
+    ContractsTree0  = aec_trees:contracts(Trees0),
+    ContractsTree1  = aect_state_tree:insert_contract(Contract, ContractsTree0),
+    Trees1          = aec_trees:set_contracts(Trees0, ContractsTree1),
+    {ContractPubKey, Contract, Trees1}.
+
+-spec run_new(aect_contracts:id(), aect_call:call(), binary(),
+              non_neg_integer(), aec_trees:trees())
+    -> aec_trees:trees().
+run_new(ContractPubKey, Call, CallData, Round, Trees0) ->
+    ContractsTree  = aec_trees:contracts(Trees0),
+    Contract = aect_state_tree:get_contract(ContractPubKey, ContractsTree),
+    Owner = aect_contracts:owner(Contract),
+    Code = aect_contracts:code(Contract),
+    CallStack = [], %% TODO: should we have a call stack for create_tx also
+                    %% when creating a contract in a contract.
+    VmVersion = aect_contracts:vm_version(Contract),
+    CallDef = #{ caller     => Owner
+               , contract   => ContractPubKey
+               , gas        => 10000000
+               , gas_price  => 1
```

Comment:
Gas is not used but we probably need it in order to verify the gas consumption does no make the contract non-postable on-chain because of high gas cost. Same goes with gas price in `run/7`

Probably I shall expose this to API? Shall it be optional as well?